File: 0933-number-of-recent-calls/0933-number-of-recent-calls.java
class RecentCounter {
    int counter;
    Queue<Integer> requests;

    public RecentCounter() {
        this.requests = new LinkedList();
    }
    
    public int ping(int t) {
        this.requests.add(t);

        int index = 0;
        Iterator<Integer> iterator = this.requests.iterator();
        while (iterator.hasNext()){
            int element = iterator.next();
            
            if (element >=  t - 3000) {
                return this.requests.size() - index;
            }

            index += 1;
        }

        return 1;
    }
}

/**
 * Your RecentCounter object will be instantiated and called as such:
 * RecentCounter obj = new RecentCounter();
 * int param_1 = obj.ping(t);
 */

File: 0933-number-of-recent-calls/0933-number-of-recent-calls.go
type RecentCounter struct {
    queue []int
}


func Constructor() RecentCounter {
    return RecentCounter{
        queue: []int{},
    }
}


func (this *RecentCounter) Ping(t int) int {
    this.queue= append(this.queue, t)
    
    i := 0 
    for this.queue[i] < t - 3000 {
        i += 1
    }

    this.queue = this.queue[i:]

    return len(this.queue)
}


/**
 * Your RecentCounter object will be instantiated and called as such:
 * obj := Constructor();
 * param_1 := obj.Ping(t);
 */